3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/adapters/registry/IgluAdapter.scala

```diff
@@ -66,21 +129,32 @@ object IgluAdapter extends Adapter {
   def toRawEvents(payload: CollectorPayload)(implicit resolver: Resolver): ValidatedRawEvents = {
 
     val params = toMap(payload.querystring)
+
+    // check if params are empty
     if (params.isEmpty) {
       "Querystring is empty: no Iglu-compatible event to process".failNel
     } else {
       params.get("schema") match {
         case None => "Querystring does not contain schema parameter: not an Iglu-compatible self-describing event".failNel
         case Some(schemaUri) => SchemaKey.parse(schemaUri) match {
           case Failure(procMsg) => procMsg.getMessage.failNel
-          case Success(_)       =>
-            NonEmptyList(RawEvent(
-              api          = payload.api,
-              parameters   = toUnstructEventParams(TrackerVersion, (params - "schema"), schemaUri, IgluFormatter, "app"),
-              contentType  = payload.contentType,
-              source       = payload.source,
-              context      = payload.context
-              )).success
+          case Success(_)       => (payload.body, payload.contentType) match {
+            case(None, _) => // if it's a GET request
+                  NonEmptyList(RawEvent(
+                    api          = payload.api,
+                    parameters   = toUnstructEventParams(TrackerVersion, (params - "schema"), schemaUri, IgluFormatter, "app"),
+                    contentType  = payload.contentType,
+                    source       = payload.source,
+                    context      = payload.context
+                    )).success
+            case(_, None) => "Content type has not been specified".failNel // POST request with no content type
+            case (Some(body), Some(contentType)) => payload.contentType match { // POST request with body and content type
+              case Some("application/json") =>
```

Comment:
A lot of webhooks end up sending a querystring within the payload.body rather than a properly formatted JSON.  To this end we should include a "application/x-www-form-urlencoded" content-type as well.